### PR TITLE
feat: Allow setting container port via PORT environment variable

### PR DESCRIPTION
**Pull request by mterrel**
Fixes adaptable/io#610

This fixes the longstanding bug that setting `PORT` in your runtime environment causes a deploy error. There could potentially be a template-independent fix done in the api, but this is super simple.

This enables us to change our onboarding and instructions to not necessarily require code changes to look at `PORT` and just allow the user to set it to a constant value instead.

### CODE REVIEW

File: index.tsx [hunk 1]
```diff
@@ -124,7 +124,15 @@ function App() {
...
     };
-    const env = mergeEnvSimple(dbEnv, standardEnv, config.environment);
+
+    // Remove PORT from the container environment
+    const { PORT, ...env } = mergeEnvSimple(dbEnv, standardEnv, config.environment) || {};
+
+    let port = 80;
+    if (PORT) {
+        const p = parseInt(PORT, 10);
+        if (!Number.isNaN(p)) port = p;
```
Comment (mvachhar) [inline index.tsx]: parseInt will parse only leading characters.  So if PORT is "10monkeys" you still get a PORT of 10.  Do we care to fix this?

**Review by mvachhar — CHANGES_REQUESTED** [review]

File: index.tsx [hunk 2]
```diff
@@ -124,7 +124,20 @@ function App() {
...
+    // Remove PORT from the container environment
+    const { PORT, ...env } = mergeEnvSimple(dbEnv, standardEnv, config.environment) || {};
+
+    let port = 80;
+    if (PORT) {
+        const msg = `Invalid value '${PORT}' for PORT in Runtime Environment`;
+        const m = /^\s*(\d+)\s*$/.exec(PORT);
+        if (!m) throw new Error(msg);
+        const p = parseInt(m[1], 10);
+        if (!Number.isNaN(p) || p < 1 || p > 65535) throw new Error(msg);
```
Comment (mvachhar) [inline index.tsx]: I believe this should be Number.isNan(p)

**Review by mvachhar — CHANGES_REQUESTED** [review]

File: index.tsx (hunk 2 above)
Comment (mvachhar) [reply index.tsx]: Sorry for the confusion, should this be `Number.isNaN(p)` instead of `!Number.isNaN(p)`

**Review by mvachhar — APPROVED** [review]

File: index.tsx (hunk 2 above)
Comment (mterrel) [reply index.tsx]: Fixed

**Review by mvachhar — APPROVED** [review]

**Pull request merged**